### ISSUE TITLE
funannotate compare poff 

### ISSUE THREAD

**Issue by estolle** [nextgenusfs/funannotate#512]
Hi

I was trying out the compare script for te first time and apparently ran into an issue that the "ProteinOrtho5 POFF results" (poff.tsv) was not found, hence it crashed. Is this a file I am supposed to supply or funannotate generates?

[11:43 AM]: OS: Ubuntu 16.04, 112 cores, ~ 528 GB RAM. Python: 3.7.8
[11:43 AM]: Running 1.8.1
[11:43 AM]: Now parsing 4 genomes
[11:43 AM]: working on Melipona beecheii
[11:44 AM]: working on Tetragonula carbonaria
[11:45 AM]: working on Scaptotrigona bipunctata
[11:46 AM]: working on Nannotrigona perilampoides
[11:47 AM]: No secondary metabolite annotations found
[11:47 AM]: Summarizing PFAM domain results
[11:47 AM]: Summarizing InterProScan results
[11:47 AM]: Loading InterPro descriptions
[11:47 AM]: Summarizing MEROPS protease results
[11:47 AM]: found 21/113 MEROPS familes with stdev >= 1.000000
/home/ek/.conda/envs/funnotate181/lib/python3.7/site-packages/funannotate/library.py:7688: MatplotlibDeprecationWarning: Calling add_axes() without argument is deprecated since 3.3 and will be removed two minor releases later. You may want to use add_subplot() instead.
  cbar_ax = fig.add_axes(shrink=0.4)
[11:47 AM]: Summarizing CAZyme results
[11:47 AM]: found 7/79 CAZy familes with stdev >= 1.000000
/home/ek/.conda/envs/funnotate181/lib/python3.7/site-packages/funannotate/library.py:7688: MatplotlibDeprecationWarning: Calling add_axes() without argument is deprecated since 3.3 and will be removed two minor releases later. You may want to use add_subplot() instead.
  cbar_ax = fig.add_axes(shrink=0.4)
[11:47 AM]: Summarizing COG results
[11:48 AM]: Summarizing secreted protein results
[11:48 AM]: Summarizing fungal transcription factors
/home/ek/.conda/envs/funnotate181/lib/python3.7/site-packages/funannotate/library.py:7688: MatplotlibDeprecationWarning: Calling add_axes() without argument is deprecated since 3.3 and will be removed two minor releases later. You may want to use add_subplot() instead.
  cbar_ax = fig.add_axes(shrink=0.4)
[11:48 AM]: Running GO enrichment for each genome
Traceback (most recent call last):
  File "/home/ek/.conda/envs/funnotate181/bin/funannotate", line 688, in <module>
    main()
  File "/home/ek/.conda/envs/funnotate181/bin/funannotate", line 678, in main
    mod.main(arguments)
  File "/home/ek/.conda/envs/funnotate181/lib/python3.7/site-packages/funannotate/compare.py", line 851, in main
    shutil.copyfile(args.proteinortho, protOrthoTSV)
  File "/home/ek/.conda/envs/funnotate181/lib/python3.7/shutil.py", line 120, in copyfile
    with open(src, 'rb') as fsrc:
FileNotFoundError: [Errno 2] No such file or directory: 'test.poff.tsv'

**Comment by estolle** [comment]
I guess its to be supplied. Could the orthology information also come from a different tool such as orthofinder?

**Comment by nextgenusfs** [comment]
It should be running proteinortho, so it seems that tool died. Any other info in the logfile?

**Comment by estolle** [comment]
hmm, not sure where to look. the compare.log ends with GO enrichment. 

The protortho folder contains the fa/faa/gff files.

I installed proteinortho via conda (not sure if it was part of the conda env before) and try rerunning now. 

Another Q. How important are the mrops/Cazy/Interproscan/SignalP annotations added via the funannotate annotate step for the compare analysis? I have a bunch of gbk files I generated directly via the funannotate util (gff2tbl and tbl2gbk). The input is from ncbi originally so they don't contain the same type of annotations. I did this (instead of funannotate annotate) because it was way faster.

**Comment by nextgenusfs** [comment]
So in the logfile there should be the command that was passed to run proteinortho -- you can try to re-run that manually and see if it spits out an error. I've run into some problems with recent versions of proteinortho as well, here was the issue https://gitlab.com/paulklemm_PHD/proteinortho/-/issues/39.  What worked on my centOS system was v6.0.14 and v6.0.16.

The annotations are all pulled from the GBK file, so if they were not annotated with `funannotate annotate` then they won't have the annotations in the GBK file where it is looking for them. Honestly, I'm not sure how useful `funannotate compare` is -- it probably needs a re-write/upgrade -- it was essentially the analyses I did for a paper several years ago.

**Comment by estolle** [comment]
no command listed in the logfile these are the last 3 entries
[11/10/20 01:21:09]: Summarizing secreted protein results
[11/10/20 01:21:09]: Summarizing fungal transcription factors
[11/10/20 01:21:11]: Running GO enrichment for each genome

I try to manually run it, lets see

**Comment by estolle** [comment]
Hi again

Ok, false alarm. Its working. I had the parameter --proteinortho test.poff.tsv and I saw in the code now that this parameter is for precomputed poff results. in the commandline help displayed for funannotate compare it says "poff results" and I thought this is an output files

Commandline help says this: 
  --proteinortho      Proteinortho POFF results. in TSV format.

in the script I found this:
    parser.add_argument('--proteinortho',
                        help='Pre-computed ProteinOrtho POFF')
